### DISCUSSION TITLE
disable floating window

### DISCUSSION THREAD

**Discussion by Masber** [ibhagwan/fzf-lua#1478] [Q&A] [answered]
Dear fzf-lua community,

I am playing with fzf-lua plugin for neovim, I was wondering whether it would be possible to disable the floating window?

I am looking for similar functionality to this vim command `let g:fzf_layout = { 'down': '40%' }`

Thank you

**Comment by ibhagwan** [comment, marked as answer]
I’m not sure if it’s the same as the fzf layout you mentioned but if I understand correctly you want to open in a bottom split?

That’s very simple with fzf-lua, set `winopts.split` to whatever command you want that opens a new window, for example:
```lua
require'fzf-lua'.setup({ winopts = { split = 'belowright new' } })
```

This can be tested with single commands too:
```lua
:lua require'fzf-lua'.files({ winopts = { split = 'belowright new' } })
```

Btw if you want `40%`, that’s possible by using winopts as a lua function and resizing the split based on `0.4*vim.o.lines`.

**Reply by xvvhang** [reply]
Hi, Is it possible to open fzf in a bottom window with no split? I've tested fzf.vim and found it works well with `vim.g.fzf_layout = { down = '~40%'}`. It would be nice to have it in fzf-lua. Thank you!

https://github.com/user-attachments/assets/079c81df-868c-4a42-964b-dff9fe2ae844

**Reply by ibhagwan** [reply]
> Hi, Is it possible to open fzf in a bottom window with no split? I've tested fzf.vim and found it works well with vim.g.fzf_layout = { down = '~40%'}. It would be nice to have it in fzf-lua. Thank you!

Thats a split in the video you’re showing, neovim can have float or split, there are no other window types.

**Reply by xvvhang** [reply]
Oh, I see. It's a botright split. Thank you very much!

**Reply by ibhagwan** [reply]
> Oh, I see. It's a botright split. Thank you very much!

If you need different size set split command to `botright split | resize …` or any other command that modifies size.

**Reply by xvvhang** [reply]
I don’t know if there an option to create a botright spilt to make fzf-lua at the bottom with a full width. That’s would be nice when there’s already horizontal spilts otherwise the fzf-lua window would be too small to show previews.

**Reply by xvvhang** [reply]
It seems like `vim.cmd[[wincmd J]]` with `on_create` works.
